src/compiler/checker.ts

```diff
@@ -6630,17 +6662,45 @@ namespace ts {
                     const targetName = getInternalSymbolName(target, verbatimTargetName);
                     includePrivateSymbol(target); // the target may be within the same scope - attempt to serialize it first
                     switch (node.kind) {
+                        case SyntaxKind.VariableDeclaration:
+                            // commonjs require: const x = require('y')
+                            if (isPropertyAccessExpression((node as VariableDeclaration).initializer!)) {
+                                // const x = require('y').z
+                                const initializer = (node as VariableDeclaration).initializer! as PropertyAccessExpression; // require('y').z
+                                const uniqueName = factory.createUniqueName((getExternalModuleRequireArgument(node) as StringLiteral).text); // _y
+                                const specifier = getSpecifierForModuleSymbol(target.parent || target, context); // 'y'
+                                // import _y = require('y');
+                                addResult(factory.createImportEqualsDeclaration(

```

Comment:
So, this always needs to get made in the outermost scope, rather than the current (potentially nested) one, since `import x = require('whatever')` isn't valid syntax within a namespace. `includePrivateSymbol` handles that for specific symbols now (by adding the symbol we want to serialize to the private symbol list for the outermost scope), however we don't have a good mechanism (currently) for adding parts of serialized results to differing scopes. You might just need a sidechannel in the `NodeBuilderContext` that you add this declaration do that you then grab and append to the start/end (probably doesn't matter which) of the file once you're doing serializing all symbols.

Reply:
As for a test case for this... something like a `module.exports.A.B.C` export that relies on a `require`'d type alias?